src/adaptors/yjs-ndn-adaptor.ts

```diff
@@ -96,9 +115,117 @@ export class NdnSvsAdaptor {
+      await this.publishUpdate(this.topic, content);
+    }
+  }
+
+  // Adam Chen Injection point 1
+  private async publishUpdate(topic: string, content: Uint8Array) {
+    await this.syncAgent.publishUpdate(topic, content);
+    // await new Promise(r => setTimeout(r,500));
+    // forced wait so that publishUpdate() is completed before we check SV.
+    console.log('-- Injection point 1: Check StateVector / Create Snapshot --');
+    const stateVector = this.syncAgent.getUpdateSyncSV();
+    console.log('debug: stateVector object: ', stateVector);
+    let count = 0;
+    for (const [_id, seq] of stateVector) {
+      count += seq;
+    }
+    console.log('Total count of state vector', count);
+    console.log('The above number should match the state vector in the debug page');
+    if (count % 5 == 0) {
+      console.log("It's time to make a snapshot!");
+      console.log('debug: group prefix: ', this.syncAgent.appPrefix.toString());
+      const encodedSV = Encoder.encode(stateVector);
+      const snapshotPrefix = this.syncAgent.appPrefix.append('32=snapshot');
+      // New SVS encodings
+      const snapshotName = snapshotPrefix.append(new Component(Version.type, encodedSV));
+      console.log('debug: targeted snapshot Prefix (persistStore key): ', snapshotPrefix.toString());
+      // /groupPrefix/32=snapshot/
+      console.log('debug: targeted snapshot Name: ', snapshotName.toString());
+      // /groupPrefix/32=snapshot/54=<stateVector>
+      const decodedSV = Decoder.decode(snapshotName.at(-1).value, StateVector);
+      console.log('debug: decoding encoded SV from snapshotName: ', decodedSV);
+      let count = 0;
+      for (const [_id, seq] of decodedSV) {
+        count += seq;
+      }
+      console.log('debug: decoding encoded SV total packet count: ', count);
+      console.log('This should match the state vector in the debug page and the previous count before encoding');
+
+      const content = Y.encodeStateAsUpdate(this.doc);
+      // its already in UTF8, transporting currently without any additional encoding.
+      console.log('yjs backend data: ', content);
+
+      // use syncAgent's blob and publish mechanism - use a different topic.
+
+      await this.syncAgent.publishBlob('snapshot', content, snapshotName, true);
+
+      //first segmented object is at /50=%00
+      const firstSegmentName = snapshotName.append('50=%00').toString();
+      console.log('debugTargetName: ', firstSegmentName);
+      const firstSegmentPacketEncoded = await this.syncAgent.persistStorage.get(firstSegmentName);
+      if (firstSegmentPacketEncoded) {
+        const firstSegmentPacket = Decoder.decode(firstSegmentPacketEncoded, Data);
+        console.log('persistentStore check: ', firstSegmentPacket);
+        console.log('persistentStore check Data Name:', firstSegmentPacket.name.toString());
+        await this.syncAgent.persistStorage.set(snapshotPrefix.toString(), Encoder.encode(firstSegmentPacket));
```

Comment:
The storage/sync-agent should provide an interface to store these kind of local stuff.
But fine for now.

Reply:
I agree. In addition, accessing the internal persistStorage of syncAgent is sort of risky and may interfere / run into race condition with regular syncAgent operation (hence why I set a couple of delays on certain sections of the code). Thanks, 'asynchronous compute'. 

Moreover, the targets (keys) of these persistStorage read/writes are hard coded and may be changed if we decide to change naming convention. Will label them out.

Perhaps we should discuss these further.

Reply:
They are part of the refactoring job: https://github.com/UCLA-IRL/ndn-workspace-solid/issues/7 and https://github.com/UCLA-IRL/ndn-workspace-solid/issues/2